### PR TITLE
#164069237 users should be able to bookmark articles for reading later

### PR DESCRIPTION
**Pull request by Peace-Apple**
## What does this PR do?

This PR adds a feature that enables a user to bookmark an article for reading later, view a list of their bookmarks and delete a bookmarked article from their list.

## Description of the task to be completed

- Enable a user to bookmark an article
- Enable a user to view their bookmark list
- Enable a user to unbookmark an article

## How should this be manually tested?

- Clone this repository
- cd into the directory Ah-backend-aquaman
- Checkout the branch ft-bookmark-article-164069237
- Create and activate your virtual environment
- pip install -r requirements.txt
- python manage.py makemmigrations
- python manage.py migrate
- Source your .env file
- python manage.py runserver
- Sign up, verify your email and login
- To bookmark an article navigate to the endpoint http://127.0.0.1:8000/api/articles/articleslug/bookmark/ and make a post request
- To retrieve a list of your bookmarks go to http://127.0.0.1:8000/api/bookmarks/ and make a get request
- To delete a bookmarked article from your bookmarks list http://127.0.0.1:8000/api/articles/articleslug/bookmark/ and make a delete request

## What are the relevant pivotal tracker stories?

[#164069237](https://www.pivotaltracker.com/story/show/164069237)

## Screenshots

Ensure to paste your auth token into the token field for all the following endpoints

To bookmark an article, copy the article slug and attach it to your URL as shown

image

To get a list of your bookmarks

image

To unbookmark an article

image

### CODE REVIEW

File: authors/apps/articles/models.py [hunk 1]
```diff
@@ -123,3 +125,10 @@ class Rating(models.Model):
 
     class Meta:
         ordering = ["-score"]
+
+
+class Bookmarks(models.Model):
```
Comment (CryceTruly) [inline authors/apps/articles/models.py]: Thanks for the good work @Peace-Apple, However, Django models are just Python classes, so the Python naming conventions detailed in PEP-8 apply.
The class Bookmarks should be named as Bookmark.

**Review by wasibani-roy — CHANGES_REQUESTED** [review]
Good work @Peace-Apple however in your PR you are missing a section. That section is the Description of Task to be completed section

**Comment by Peace-Apple** [comment]
> Good work @Peace-Apple however in your PR you are missing a section. That section is the Description of Task to be completed section

Okay thanks Roy, let me add that

**Review by CryceTruly — CHANGES_REQUESTED** [review]

File: authors/apps/articles/models.py (hunk 1 above)
Comment (Peace-Apple) [reply authors/apps/articles/models.py]: Thanks, @CryceTruly  for pointing that out, I have corrected it

**Review by CryceTruly — DISMISSED** [review]

**Review by wasibani-roy — APPROVED** [review]

**Review by CryceTruly — APPROVED** [review]

**Review by BagzieGracious — APPROVED** [review]

**Pull request merged**